zarp/run/snakemake.py

```diff
@@ -0,0 +1,49 @@
+"""Module for running snakemake."""
+
+import subprocess
+import os
+
+from zarp.config.models import Run
+
+
+class SnakemakeExecutor:
+    """Run snakemake with system calls."""
+
+    def __init__(self, run: Run) -> None:
+        self.run_dict = run.dict()
+        self.success = None
+
+    def prepare_run(self, snkfile: str, workdir: str) -> None:
+        """Configure list of strings for execution.
+        """
+        run_list = ["snakemake"]
+        # Snakemake config
+        run_list.extend(["--snakefile", snkfile])
+        run_list.extend(["--cores", str(self.run_dict["cores"])])
+        # run_list.extend(["--config", f"workdir={workdir}"])
+        # execution profile
+        if self.run_dict['execution_profile'] == "local-conda":
+            prof = ["--profile", os.path.join("submodules", "zarp", "profiles", "local-conda")]
+            run_list.extend(prof)
+        self.run_list = run_list
+
+    def validate_run(self) -> bool:
+        """Ensure the list of strings is a correct Snakemake call.
+        """
+        pass
+
+    def run(self) -> None:
+        """Execute Snakemake with system call."""
+        try:
+            completed = subprocess.run(self.run_list)
+            if completed.returncode == 0:
+                print("Successfully finished!")
+                self.success = True
+            else:
+                print(f"Snakemake not finished: {completed}")
+                self.success = False
+        except subprocess.CalledProcessError as e:
+            print(e)
```

Comment:
Note that you are just printing out the error message here, then ZARP-cli continues. It's probably better to raise this error. And you may wanna keep track of STDERR and print _that_. Similar to here: https://github.com/zavolanlab/htsinfer/blob/cf162d5ff8b7b2988d2bd4fb0c799a032de3ce5b/htsinfer/get_read_orientation.py#L208-L215

Although it _is_ a good idea to wrap the code in a `try` - `catch` for `CalledProcessError` as you do.

So I would:
- check the return code and print STDERR if it's non-zero
- catch `CalledProcessError`
- in both caises raise the error to the calling context (don't forget to document that in the `Raises` section in the docstring!)

Note that you can reraise exceptions in Python3. The pattern is:

```
except SomeException as exc:
    raise SameExceptionOrAnotherOne("Some descriptive error message") from exc
```

So the interesting part here is the `from`. This way you will keep all information about the original error but _also_ have the ability to add your own stuff to the error message/traceback.

Reply:
the variable `completed` contains all information, including STDERR. 
So one LOGGING is included, that could be passed. But not sure if it should raise an error. At least, if it is, the attribute `success` could be removed.

Reply:
Ok, I changed it such that [`subprocess`](https://docs.python.org/3/library/subprocess.html#module-subprocess) will raise a `CalledProcessError` when exit code != 0. I think this it the intended way. 
I still set the attribute `success` but I don't think it has a real use case here. Anyway, I'd like to keep it for now, to not loose myself in details :-).